File: PostProcessing/js/Runtime/Effects/VignetteRenderer.js
const { Utils } = require("PostProcessing/Base/Utils");
const { PostProcessBaseRenderer } = require("PostProcessing/Runtime/Effects/PostProcessBaseRenderer")
const Amaz = effect.Amaz;

// uniform names
const u_power = "u_power"
const u_contrast = "u_contrast"
const u_ScreenParams = "u_ScreenParams"
//I'm not sure how to set these
//u_image;
//u_maskA;


// pass ids
const VignettePass = 0

class VignetteRenderer extends PostProcessBaseRenderer{
    constructor()
    {
        super();
        this.commands = new Amaz.CommandBuffer();

        this.setRendererProperty("pVignetteEnable", false, true);
        this.setRendererProperty("pVignettePower", 0.00);
        this.setRendererProperty("pVignetteContrast", 1.00);
    }

    update(properties)
    {
        this.updateProperties(properties);
    }
    
    render(scene,postProcessContext)
    {
        //  if (postProcessContext  === null){
        //     return;
        //  }
        var enable = this.settings.get("pVignetteEnable");

        var power = this.settings.get("pVignettePower");
        var contrast = this.settings.get("pVignetteContrast");

        var width = postProcessContext.getScreenWidth();
        var height = postProcessContext.getScreenHeight();

        if (enable)
        {
            var VignetteMat = postProcessContext.getResources().getShaders("Vignette").getMaterial();

            // Set Vignette uniform
            VignetteMat.setFloat(u_power, power);
            VignetteMat.setFloat(u_contrast, contrast);

            if (this.dirty)
            {
                var w = width;
                var h = height;
                var src = postProcessContext.getSource();
                var dst = postProcessContext.getDestination();
                if (src.image === dst.image)
                {
                    var pingpong = Utils.CreateRenderTexture("", width, height);
                    this.commands.blitWithMaterial(src, pingpong, VignetteMat, VignettePass);
                    this.commands.blit(pingpong, dst);
                }else{
                    this.commands.blitWithMaterial(src, dst, VignetteMat, VignettePass);
                }
                this.dirty = false;
            }
        }else{
            if (this.dirty)
            {
                this.commands.clearAll();
                this.dirty = false;
                // to do, if (src !== dst, copy src to dst
            }
        }
        //}

        var cam = postProcessContext.getCamera();
        cam.entity.scene.commitCommandBuffer(this.commands);
    }
}

exports.VignetteRenderer = VignetteRenderer;

File: PostProcessing/lua/Runtime/Effects/VignetteRenderer.lua
VignetteRenderer = ScriptableObject(PostProcessEffectRenderer)

-- uniform names
local u_power = "u_power"
local u_contrast = "u_contrast"
local u_ScreenParams = "u_ScreenParams"
--I'm not sure how to set these
--u_image;
--u_maskA;


-- pass ids
local VignettePass = 0

function VignetteRenderer : ctor()
end

function VignetteRenderer : render(sys,renderContext)
    --  if renderContext  == nil then
    --     return
    --  end
    local enable = self.settings["pVignetteEnable"]
    local settings = self.settings

    local power = settings["pVignettePower"]
    local contrast = settings["pVignetteContrast"]

    local commands = renderContext:getCommandBuffer()
    local width = renderContext:getScreenWidth()
    local height = renderContext:getScreenHeight()

    if enable then
        local VignetteMat = renderContext:getResources():getShaders("Vignette"):getMaterial()

        -- Set Vignette uniform
        VignetteMat:setFloat(u_power, power)
        VignetteMat:setFloat(u_contrast, contrast)

        if self.dirty then
            local w = width
            local h = height
            local src = renderContext:getSource()
            local dst = renderContext:getDestination()
            if src.image == dst.image then
                local pingpong = CreateRenderTexture("", width, height)
                commands:blitWithMaterial(src, pingpong, VignetteMat, VignettePass)
                commands:blit(pingpong, dst)
            else
                commands:blitWithMaterial(src, dst, VignetteMat, VignettePass)
            end
            self.dirty = false
        end
    else
            if self.dirty then
                commands:clearAll()
                self.dirty = false
                -- to do, if src ~= dst, copy src to dst
            end
    end
    --end

    local cam = renderContext:getCamera()
    cam.entity.scene:commitCommandBuffer(commands)
end